egograph/backend/infrastructure/database/timeline_queries.py

```
@@ -0,0 +1,254 @@
+    sql = f"""
+        SELECT
+            {_TS.format(col="started_at_utc")} AS started_at_utc,
+            {_TS.format(col="ended_at_utc")} AS ended_at_utc,
+            duration_seconds,
+            session_type
+        FROM read_parquet(?)
+        WHERE data_type = 'sleep'
+          AND (({_TS.format(col="ended_at_utc")} AT TIME ZONE 'UTC')
+                AT TIME ZONE '{params.tz_name}')::DATE = ?
+        ORDER BY started_at_utc ASC
+    """
```

Ruff (0.15.20)

[error] Possible SQL injection vector through string-based query construction

(S608)